packages/ember-glimmer/lib/helpers/readonly.js

```diff
@@ -5,6 +5,101 @@
 import { UPDATE } from '../utils/references';
 import { unMut } from './mut';
 
+
+/**
+  The `readonly` helper let's you specify that when you updated a value passed in
+  to a child component, it does not update in the parent component.
+  
+  To specify that a parameter is read-only, when invoking the child `Component`:
+  
+    ```javascript
+  // my-parent.js
+  export default Component.extend({
+    totalClicks: 3
+  });
+  ```
+  
+  ```handlebars
+  {{log totalClicks}} // -> 3
+  {{my-child childClickCount=(readonly totalClicks)}}
+  ```
+  
+  Now, when you update `childClickCount`:
+  
+  ```javascript
+  // my-child.js
+  export default Component.extend({
+    click() {
+      this.incrementProperty('childClickCount');
+    }
+  });
+  ```
+  
+  The value updates in the child component, but not the parent component:
+  
+  ```handlebars
+  // my-child.hbs
+  {{log childClickCount}} //-> 4
+  ```
+  
+  ```handlebars
+  // my-parent.hbs
+  {{log totalClicks}} //-> 3
+  {{my-child childClickCount=(readonly totalClicks)}}
+  ```
+  
+  ### Objects and Arrays
+  
+  Note that while properties passed in as `readonly` parameters won't update
+  the property itself, if it is an Object or an Array, you can still modify
+  properties of it and it'll update in the parent. You can think of this as
+  working similarly to the `const` keyword introduced in ES6. Let's look at
+  an example:
```

Comment:
What do you think about:

When passing property that is a complex object vs. a primitive object, only the reference to the object is protected using the `readonly` helper. The contents of a bound property that is an `Object` or `Array` can be changed in both the `outerProperty` and the `componentProperty` simultaneously. This read only (or one-way) binding behaves similar to the `const` keyword in ES2015.

See the following example:

*Suggestion: use "ES2105" instead of "ES6" (ECMAScript 6 (ES6)/ ECMAScript 2015 (ES2015): The 6th edition of ECMAScript, standardized in 2015)*